### PR TITLE
Use the cordova lifecycle hooks to build stuff

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

**Comment by jbuck** [comment]
@k88hudson r?

**Comment by jbuck** [comment]
@thisandagain r?

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,7 @@
 .DS_Store
 
-build/
 platforms/
 plugins/
-www/
```
Comment (k88hudson) [inline .gitignore]: why did you remove this?

File: .travis.yml [hunk 2]
```diff
@@ -1,29 +1,19 @@
...
-- sudo add-apt-repository -y ppa:chris-lea/node.js
-- sudo apt-get update -y
-- sudo apt-get install -y nodejs
+- curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add -
+- echo 'deb https://deb.nodesource.com/node precise main' | sudo tee /etc/apt/sources.list.d/nodesource.list > /dev/null
+- sudo apt-get -y update
+- sudo apt-get -y install nodejs
 - sudo npm -g install cordova
 - sudo npm install
+- npm start
```
Comment (k88hudson) [inline .travis.yml]: you need to do npm build as well

File: README.md [hunk 3]
```diff
@@ -1,12 +1,10 @@
-### Webmaker App Cordova
```
Comment (k88hudson) [inline README.md]: we should actually move this to the main documentation in webmaker-app

Comment (k88hudson) [inline README.md]: im ok with that being a separate ticket though

File: .gitignore (hunk 1 above)
Comment (jbuck) [inline .gitignore]: I added a .gitignore file in www, because cordova requires that the www directory exists, but I don't want to actually version control anything in there

File: package.json [hunk 4]
```diff
@@ -4,12 +4,11 @@
   "description": "[![Build Status](https://travis-ci.org/mozillafordevelopment/beaker.svg)](https://travis-ci.org/mozillafordevelopment/beaker)",
   "main": "index.js",
   "scripts": {
-    "start": "node ./scripts/build.js && cordova prepare && cordova run android",
-    "build": "node ./scripts/build.js && cordova prepare",
-    "android:prepare": "node ./scripts/copy-assets.js --android && cordova prepare",
-    "android:build": "npm run android:prepare && cordova build android",
-    "android": "npm run android:prepare && cordova run android",
-    "firefoxos": "node ./scripts/copy-assets.js && cordova prepare firefoxos"
+    "start": "npm run add-platform && echo 'Now type `npm run android` or `npm build firefoxos` to run webmaker-app on your desired platform'",
```
Comment (k88hudson) [inline package.json]: Nice! I like the message at the end

File: .travis.yml (hunk 2 above)
Comment (jbuck) [inline .travis.yml]: I do the `npm run build` in the script step

File: package.json (hunk 4 above)
Comment (jbuck) [inline package.json]: oops, just realized I added the wrong npm script here though

File: README.md (hunk 3 above)
Comment (jbuck) [inline README.md]: yarp

File: scripts/after_platform_add.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+var async = require('async');
+var spawn = require('./spawn');
+
+var addPlugin = function(plugin, callback) {
+    spawn('cordova', ['plugin', 'add', plugin], callback);
+};
+
+module.exports = function(context) {
+    var Q = context.requireCordovaModule('q');
```
Comment (k88hudson) [inline scripts/after_platform_add.js]: this is very interesting lol

File: .gitignore (hunk 1 above)
Comment (k88hudson) [inline .gitignore]: ...and use mkdirp to create the www directory

File: www/.gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
```
Comment (k88hudson) [inline www/.gitignore]: see above

File: package.json (hunk 4 above)
Comment (k88hudson) [inline package.json]: haha did you mean to add build?

File: scripts/after_platform_add.js (hunk 5 above)
Comment (jbuck) [inline scripts/after_platform_add.js]: yeah, it's the cordova recommended way of having async hooks... and it's freaking weird

File: .travis.yml (hunk 2 above)
Comment (k88hudson) [inline .travis.yml]: ah! sorry, missed it

**Comment by k88hudson** [comment]
A few nits but this is :star2: :star2: :star2: :star2: :star2: :star2: :star2: :star2:

File: scripts/after_platform_add.js (hunk 5 above)
Comment (thisandagain) [inline scripts/after_platform_add.js]: wat

Comment (jbuck) [inline scripts/after_platform_add.js]: I swears it: https://github.com/apache/cordova-lib/blob/master/cordova-lib/templates/hooks-README.md#javascript
